openml/runs/run.py

```diff
@@ -186,24 +192,45 @@ def _generate_arff_dict(self):
         class_labels = task.class_labels
 
         arff_dict = OrderedDict()
-        arff_dict['attributes'] = [('repeat', 'NUMERIC'),  # lowercase 'numeric' gives an error
-                                   ('fold', 'NUMERIC'),
-                                   ('sample', 'NUMERIC'),
-                                   ('row_id', 'NUMERIC')] + \
-            [('confidence.' + class_labels[i], 'NUMERIC') for i in range(len(class_labels))] +\
-            [('prediction', class_labels),
-             ('correct', class_labels)]
         arff_dict['data'] = self.data_content
         arff_dict['description'] = "\n".join(run_environment)
-        arff_dict['relation'] = 'openml_task_' + str(task.task_id) + '_predictions'
+        arff_dict['relation'] = 'openml_task_' + str(task.task_id) + \
+                                '_predictions'
+
+        # Separate these out? Normal classification doesn't need 'sample'
+        if task.task_type in ['Supervised Classification', 'Learning Curve']:
+            arff_dict['attributes'] = [
+                                          ('repeat', 'NUMERIC'),
+                                          ('fold', 'NUMERIC'),
+                                          ('sample', 'NUMERIC'),
+                                          ('row_id', 'NUMERIC')] + \
+                                      [('confidence.' + class_labels[i],
+                                        'NUMERIC') for i in
+                                       range(len(class_labels))] + \
+                                      [('prediction', class_labels),
+                                       ('correct', class_labels)]
+
+        elif task.task_type == 'Supervised Regression':
+            arff_dict['attributes'] = [('repeat', 'NUMERIC'),
+                                       ('fold', 'NUMERIC'),
+                                       ('row_id', 'NUMERIC'),
+                                       ('prediction', 'NUMERIC'),
+                                       ('truth', 'NUMERIC')]
```

Comment:
why deviate from `correct`? Makes it harder to document

Reply:
This is what it said in the definition of the regression task. I didn't make this up. I thought about asking for an update to the regression task type on the server, but since it's already used in the Java/R API I guess we can live with it.